x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -464,3 +555,97 @@ type kibanaAPIKeys struct {
+	URL             string `config:"url"`
+}
+
+type fleetServerConfig struct {
+	Cert          string              `config:"cert"`
+	CertKey       string              `config:"certKey"`
+	Elasticsearch elasticsearchConfig `config:"elasticsearch"`
+	Enable        bool                `config:"enable"`
+	Host          string              `config:"host"`
+	InsecureHTTP  bool                `config:"insecure_http"`
+	PolicyID      string              `config:"policy_id"`
+	PolicyName    string              `config:"policy_name"`
+	Port          string              `config:"port"`
+}
+
+type kibanaConfig struct {
+	Fleet kibanaFleetConfig `config:"fleet"`
+}
+
+type kibanaFleetConfig struct {
+	CA       string `config:"ca"`
+	Host     string `config:"host"`
+	Password string `config:"password"`
+	Setup    bool   `config:"setup"`
+	Username string `config:"username"`
+}
+
+func defaultAccessConfig() setupConfig {
+	return setupConfig{
+		Fleet: fleetConfig{
+			CA:              envWithDefault("", "FLEET_CA", "KIBANA_CA", "ELASTICSEARCH_CA"),
+			Enroll:          envBool("FLEET_ENROLL", "FLEET_SERVER_ENABLE"),
+			EnrollmentToken: envWithDefault("", "FLEET_ENROLLMENT_TOKEN"),
+			Force:           envBool("FLEET_FORCE"),
+			Insecure:        envBool("FLEET_INSECURE"),
+			TokenName:       envWithDefault("Default", "FLEET_TOKEN_NAME"),
+			TokenPolicyName: envWithDefault("", "FLEET_TOKEN_POLICY_NAME"),
+			URL:             envWithDefault("", "FLEET_URL"),
+		},
+		FleetServer: fleetServerConfig{
+			Cert:    envWithDefault("", "FLEET_SERVER_CERT"),
+			CertKey: envWithDefault("", "FLEET_SERVER_CERT_KEY"),
+			Elasticsearch: elasticsearchConfig{
+				Host:     envWithDefault("http://elasticsearch:9200", "FLEET_SERVER_ELASTICSEARCH_HOST", "ELASTICSEARCH_HOST"),
+				Username: envWithDefault("elastic", "FLEET_SERVER_ELASTICSEARCH_USERNAME", "ELASTICSEARCH_USERNAME"),
+				Password: envWithDefault("changeme", "FLEET_SERVER_ELASTICSEARCH_PASSWORD", "ELASTICSEARCH_PASSWORD"),
+				CA:       envWithDefault("", "FLEET_SERVER_ELASTICSEARCH_CA", "ELASTICSEARCH_CA"),
+			},
+			Enable:       envBool("FLEET_SERVER_ENABLE"),
+			Host:         envWithDefault("", "FLEET_SERVER_HOST"),
+			InsecureHTTP: envBool("FLEET_SERVER_INSECURE_HTTP"),
+			PolicyID:     envWithDefault("", "FLEET_SERVER_POLICY_ID"),
+			PolicyName:   envWithDefault("", "FLEET_SERVER_POLICY_NAME", "FLEET_TOKEN_POLICY_NAME"),
+			Port:         envWithDefault("", "FLEET_SERVER_PORT"),
+		},
+		Kibana: kibanaConfig{
+			Fleet: kibanaFleetConfig{
+				// Remove FLEET_SETUP in 8.x
+				// The FLEET_SETUP environment variable boolean is a fallback to the old name. The name was updated to
+				// reflect that its setting up Fleet in Kibana versus setting up Fleet Server.
+				Setup: envBool("KIBANA_FLEET_SETUP", "FLEET_SETUP"),
+				Host:  envWithDefault("http://kibana:5601", "KIBANA_FLEET_HOST", "KIBANA_HOST"),
+				//TODO(simitt): check why ELASTICSEARCH values are used here?
```

Comment:
I assume it is as a fallback to not have to configure credentials for ES and KB as they are the same in simple setups.

Reply:
yeah, could easily be the source of confusion though.

Reply:
Correct it is a fallback to make it easier on a user. Removing the need environment variables that have the same value.